cmd/mapcidr/main.go

```diff
@@ -324,33 +333,27 @@ func process(wg *sync.WaitGroup, chancidr, outputchan chan string) {
 			continue
 		}
 
-		// Add IPs which are passed as input. Example - "192.168.0.0-192.168.0.5"
-		if options.IPRangeInput {
-			ipRange = append(ipRange, ip)
-		} else if options.Aggregate || options.Shuffle || hasSort || options.AggregateApprox || options.Count {
-			// In case of coalesce/shuffle we need to know all the cidrs and aggregate them by calling the proper function
+		// In case of coalesce/shuffle we need to know all the cidrs and aggregate them by calling the proper function
+		if options.Aggregate || options.Shuffle || hasSort || options.AggregateApprox || options.Count {
 			_ = ranger.AddIPNet(pCidr)
 			allCidrs = append(allCidrs, pCidr)
 		} else {
 			commonFunc(cidr, outputchan)
 		}
 	}
-
 	if options.IPRangeInput {
-		// check if ipRange has more than 2 values
-		if len(ipRange) > 2 {
-			gologger.Fatal().Msgf("IP range can not have more than 2 values.")
-		}
-		cidrs, err := mapcidr.GetCIDRFromIPRange(ipRange[0], ipRange[1])
-		if err != nil {
-			gologger.Fatal().Msgf("%s\n", err)
-		}
+		for _, ipRange := range ipRangeMap {
```

Comment:
You are not doing anything with the `key` of the map, so why need a map in first place?

Reply:
Remove the map with the list